torchaudio/datasets/librispeech.py

```diff
@@ -0,0 +1,81 @@
+import os
+
+import torch.utils.data as data
+
+import torchaudio
+from torchaudio.datasets.utils import download, extract, shuffle, walk
+
+
+def load_librispeech_item(fileid, path, ext_audio, ext_txt):
+
+    speaker, chapter, utterance = fileid.split("-")
+
+    file_text = speaker + "-" + chapter + ext_txt
+    file_text = os.path.join(path, speaker, chapter, file_text)
+    file_audio = speaker + "-" + chapter + "-" + utterance + ext_audio
+    file_audio = os.path.join(path, speaker, chapter, file_audio)
+
+    # Load audio
+    waveform, sample_rate = torchaudio.load(file_audio)
+
+    # Load text
+    for line in open(file_text):
+        fileid_text, content = line.strip().split(" ", 1)
+        if file_audio == fileid_text:
+            break
+    else:
+        # Translation not found
+        raise ValueError
+
+    return {
+        "speaker": speaker,
+        "chapter": chapter,
+        "utterance": utterance,
+        "content": content,
+        "waveform": waveform,
+        "sample_rate": sample_rate,
+    }
+
+
+class LIBRISPEECH(data.IterableDataset):
+
+    _ext_txt = ".trans.txt"
+    _ext_audio = ".flac"
+
+    def __init__(self, root, selection):
+
+        selections = [
+            "dev-clean",
+            "test-clean",
+            "test-other",
+            "train-clean-100",
+            "train-clean-360",
+            "train-other-500",
+        ]
+
+        assert selection in selections
+
+        ext_archive = ".tar.gz"
+        base = "http://www.openslr.org/resources/12/"
+        url = os.path.join(base, selection + ext_archive)
+        folder_in_archive = os.path.join("LibriSpeech", selection)
+
+        archive = os.path.basename(url)
+        archive = os.path.join(root, archive)
+        self._path = os.path.join(root, folder_in_archive)
+
+        if not os.path.isdir(self._path):
+            if not os.path.isfile(archive):
+                torchaudio.datasets.utils.download_url(_url, root)
+            torchaudio.datasets.utils.extract_archive(archive)
+
+        walker = torchaudio.datasets.utils.walk_files(
```

Comment:
Not sure walking here is the right approach. This is unfortunately one of those situations where the "input" to the pipeline is a list of paths that then are randomized (sometimes based on some per-path metadata). We could offer the user to provide these paths as an argument as well and reference the walk_files util as a means of retrieving them.